driver/src/logging.rs

```diff
@@ -21,47 +27,64 @@ pub fn init(filter: impl AsRef<str>) -> (Logger, GlobalLoggerGuard) {
     (logger, guard)
 }
 
-pub struct CustomFormatter<D: Decorator> {
-    decorator: D,
+/// Uses one decorator for `Error` and `Critical` log messages and the other for
+/// the rest.
+pub struct CustomFormatter<T0, T1> {
```

Comment:
```suggestion
pub struct CustomFormatter<Out, Err> {
```

Reply:
+1, nbd, but any reason this was not incorporated?

Reply:
Imo this isn't an improvement because Err is also confusing because its usually used for Results. We can go all the way and call the types `StdoutDecorator` and `StderrDecorator` if you prefer that.

Reply:
I would just use `Stdout` and `Stderr` then. I'm not a fan of the numbered generic types here.